File: spec/javascript/test_expr_2.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('Expr2', 'src/str_encodings.bin', function(r) {
    assert.equal(r.str1.lenOrig, 10)
    assert.equal(r.str1.lenMod, 7)
    assert.equal(r.str1.str, "Some AS")

    assert.equal(r.str1Len, 7)
    assert.equal(r.str1LenMod, 7)
    assert.equal(r.str1Byte1, 0x49)
    assert.equal(r.str1Avg, 0x49)
    assert.equal(r.str1Char5, "e")

    assert.equal(r.str1Tuple5.byte0, 0x65)
    assert.equal(r.str1Tuple5.byte1, 0x20)
    assert.equal(r.str1Tuple5.byte2, 0x41)
    assert.equal(r.str1Tuple5.avg, 0x30)

    assert.equal(r.str2Tuple5.byte0, 0x65)
    assert.equal(r.str2Tuple5.byte1, 0x20)
    assert.equal(r.str2Tuple5.byte2, 0x41)
    assert.equal(r.str2Tuple5.avg, 0x30)
});

File: spec/cpp_stl/test_expr_2.cpp
#include <boost/test/unit_test.hpp>

#include <expr_2.h>

#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_expr_2) {
    std::ifstream ifs("src/str_encodings.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    expr_2_t* r = new expr_2_t(&ks);

    BOOST_CHECK_EQUAL(r->str1()->len_orig(), 10);
    BOOST_CHECK_EQUAL(r->str1()->len_mod(), 7);
    BOOST_CHECK_EQUAL(r->str1()->str(), "Some AS");

    BOOST_CHECK_EQUAL(r->str1_len(), 7);
    BOOST_CHECK_EQUAL(r->str1_len_mod(), 7);
    BOOST_CHECK_EQUAL(r->str1_byte1(), 0x49);
    BOOST_CHECK_EQUAL(r->str1_avg(), 0x49);
    BOOST_CHECK_EQUAL(r->str1_char5(), "e");

    BOOST_CHECK_EQUAL(r->str1_tuple5()->byte0(), 0x65);
    BOOST_CHECK_EQUAL(r->str1_tuple5()->byte1(), 0x20);
    BOOST_CHECK_EQUAL(r->str1_tuple5()->byte2(), 0x41);
    BOOST_CHECK_EQUAL(r->str1_tuple5()->avg(), 0x30);

    BOOST_CHECK_EQUAL(r->str2_tuple5()->byte0(), 0x65);
    BOOST_CHECK_EQUAL(r->str2_tuple5()->byte1(), 0x20);
    BOOST_CHECK_EQUAL(r->str2_tuple5()->byte2(), 0x41);
    BOOST_CHECK_EQUAL(r->str2_tuple5()->avg(), 0x30);

    delete r;
}
